ecrtokencronjob/src/kubernetes/namespaces.go

```diff
@@ -0,0 +1,146 @@
+package k8s
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var namespaceWhitespace = []string{" ", "\r", "\t", "\v"}
+var namespaceSeparators = []string{"\n", ";"}
+
+func GetNamespaces(envVar string) ([]string, error) {
+	if envVar == "" {
+		envVar = "default"
+	}
+
+	envVar = formatNamespaceList(envVar)
+
+	list := strings.Split(envVar, ",")
+	single := make([]string, 0, len(list))
+	regex := make([]*regexp.Regexp, 0, len(list))
+
+	for _, val := range list {
+		if hasWildCard(val) {
+			r, err := getRegex(val)
+			if err != nil {
+				return nil, err
+			}
+			regex = append(regex, r)
+		} else {
+			single = append(single, val)
+		}
+	}
+
+	matchedNamespaces, err := findNamespaces(regex)
+	if err != nil {
+		return nil, err
+	}
+
+	return unique(append(single, matchedNamespaces...)), nil
```

Comment:
Nil: it seems to be a better practice to use a set instead. Refer to [this](https://stackoverflow.com/questions/33207197/how-can-i-create-an-array-that-contains-unique-strings) as an example.

Though given I cannot image there will be 100+ namespaces, there will probably be no performance difference. So I'm fine with skipping this change if time is a concern.